lux/core/frame.py

```diff
@@ -426,30 +428,55 @@ def maintain_recs(self, is_series="DataFrame"):
             elif not (len(rec_df) < 5 and not rec_df.pre_aggregated and not is_sql_tbl) and not (
                 self.index.nlevels >= 2 or self.columns.nlevels >= 2
             ):
-                from lux.action.custom import custom_actions
+                from lux.action.custom import custom_action, filter_keys
+
+                self.action_keys = filter_keys(rec_df, self.loadingBar)
+
+                if lux.config._streaming:
+                    # Compute one tab to display on initial widget
+                    if len(self.action_keys) > 0:
+                        rec = custom_action(rec_df, self.action_keys[0])
+                        rec_df._append_rec(rec_infolist, rec)
+                        self.action_keys.pop(0)
+
+                        # Fill the rest of the tabs with empty (non-clickable) tabs
+                        for action_name in self.action_keys:
+                            rec = {
+                                "action": action_name.capitalize(),
+                                "description": "",
+                                "long_description": "",
+                                "collection": [],
+                            }
+                            rec_infolist.append(rec)
+                else:
+                    for action_name in self.action_keys:
+                        rec = custom_action(rec_df, action_name)
+                        rec_df._append_rec(rec_infolist, rec)
+                    self.action_keys = []
 
-                # generate vis from globally registered actions and append to dataframe
-                custom_action_collection = custom_actions(rec_df)
-                for rec in custom_action_collection:
-                    rec_df._append_rec(rec_infolist, rec)
                 lux.config.update_actions["flag"] = False
 
             # Store _rec_info into a more user-friendly dictionary form
             rec_df._recommendation = {}
             for rec_info in rec_infolist:
                 action_type = rec_info["action"]
                 vlist = rec_info["collection"]
-                if len(vlist) > 0:
-                    rec_df._recommendation[action_type] = vlist
+                rec_df._recommendation[action_type] = vlist
             rec_df._rec_info = rec_infolist
             rec_df.show_all_column_vis()
-            if lux.config.render_widget:
-                self._widget = rec_df.render_widget()
+
+            self._widget = rec_df.render_widget(
+                pandasHtml=rec_df.to_html(max_rows=5, classes="pandasStyle")
```

Comment:
Let's make this 10 so that it corresponds to the default [`display.min_rows` value](https://pandas.pydata.org/pandas-docs/stable/user_guide/options.html) which is 10